test/utils/deploy/openshift.go

```diff
@@ -47,6 +47,10 @@ func (o *OpenShiftNRO) deployWithLabels(ctx context.Context, matchLabels map[str
 	kcObj, err := objects.TestKC(matchLabels)
 	Expect(err).To(Not(HaveOccurred()))
 
+	// need to get reference values before any mutations
+	mcps, err := nropmcp.GetListByNodeGroupsV1(ctx, e2eclient.Client, nroObj.Spec.NodeGroups)
+	Expect(err).NotTo(HaveOccurred())
+
 	unpause, err := e2epause.MachineConfigPoolsByNodeGroups(nroObj.Spec.NodeGroups)
```

Comment:
maybe move to `Eventually` here as well

Reply:
I think just wrapping into `Eventually` is more a band-aid fix. I'll look for a redesign of the code in a later PR.